server.go

```diff
@@ -711,7 +713,7 @@ func (sp *serverPeer) OnVersion(p *peer.Peer, msg *wire.MsgVersion) *wire.MsgRej
 		// Request known addresses if the server address manager needs
 		// more.
 		if addrManager.NeedMoreAddresses() {
-			p.QueueMessage(wire.NewMsgGetAddr(), nil)
+			sp.QueueMessage(wire.NewMsgGetAddr(), nil)
```

Comment:
Though I discuss updating the message listener function signatures considering `p` and `sp` in this modification are identical, correct me if I'm wrong. We currently have the form `OnEvent(p *Peer, msg *wire.Msg)`. Glancing through the codebase, the types implementing listeners are peer types which makes `p` redundant. If I'm not misinformed there are no objections I'd like to start working on this.

Reply:
For dcrd they are redundant, because of the fact we close over each one, but not for other applications, hence I don't think they should be removed from `peer`, because it would force everyone else using it to close over the state too when they otherwise don't need to.